src/main/java/life/qbic/omero/BasicOMEROClient.java

```diff
@@ -3,384 +3,526 @@
+   * @throws DSOutOfServiceException
+   * @throws DSAccessException
+   * @throws ExecutionException
+   */
+  public List<FileAnnotationData> fetchFileAnnotationDataForImage(long imageID)
+      throws DSOutOfServiceException, DSAccessException, ExecutionException {
+    List<FileAnnotationData> annotations = new ArrayList<>();
+    for (AnnotationData d : loadAnnotationsForImage(imageID, FileAnnotationData.class)) {
+      annotations.add((FileAnnotationData) d);
+    }
+    return annotations;
+  }
+
+  /**
+   * Returns any map annotation data (key value pairs of metadata) of a given image
+   * 
+   * @param imageID the ID of the image
+   * @return A list of MapAnnotationData objects
+   * @throws DSOutOfServiceException
+   * @throws DSAccessException
+   * @throws ExecutionException
+   */
+  public List<MapAnnotationData> fetchMapAnnotationDataForImage(long imageID)
+      throws DSOutOfServiceException, DSAccessException, ExecutionException {
+    List<MapAnnotationData> annotations = new ArrayList<>();
+    for (AnnotationData d : loadAnnotationsForImage(imageID, MapAnnotationData.class)) {
+      annotations.add((MapAnnotationData) d);
+    }
+    return annotations;
+  }
 
-            this.connect(this.hostname, this.port, this.username, this.password);
-        } catch (Exception e) {
-            System.out.println(e);
+  private Collection<? extends AnnotationData> loadAnnotationsForImage(long imageID,
+      Class<? extends AnnotationData> type)
+      throws DSOutOfServiceException, DSAccessException, ExecutionException {
+    long userID = this.gateway.getLoggedInUser().getId();
 
-            System.out.println("++++++++++++++++++++++++++++++++++++++");
-            System.out.println(e.getLocalizedMessage());
-            System.out.println(e.getMessage());
-            System.out.println(e.toString());
+    BrowseFacility browse = gateway.getFacility(BrowseFacility.class);
+    ImageData image = browse.getImage(this.ctx, imageID);
 
-        }
+    List<Long> userIds = new ArrayList<Long>();
+    userIds.add(userID);
 
-    }
+    List<Class<? extends AnnotationData>> types = new ArrayList<Class<? extends AnnotationData>>();
+    types.add(type);
 
-    public String getSessionId(){
+    MetadataFacility metadata = gateway.getFacility(MetadataFacility.class);
+    List<AnnotationData> annotations = metadata.getAnnotations(ctx, image, types, userIds);
 
-        return this.sessionId;
-    }
+    disconnect();
 
-    public void disconnect() {
-        this.gateway.disconnect();
+    if (annotations != null) {
+      return annotations;
+    }
+    return new ArrayList<>();
+  }
+
+
+  /**
+   * render buffered image of image object in omero
+   * 
+   * @param image
+   * @param zPLane
+   * @param tPlane
+   * @return
+   * @throws ServerError
+   * @throws DSOutOfServiceException
+   * @throws IOException
+   */
+  private BufferedImage render(ImageData image, int zPLane, int tPlane)
```

Comment:
```suggestion
  private BufferedImage renderOmeroImage(ImageData image, int zPLane, int tPlane)
```

Reply:
It's a real image (BufferedImage class in Java). I think calling it Omero might be misleading. RenderImage[fromOmeroData]?